pymilvus/v2/apis.py

```diff
@@ -118,13 +177,63 @@ def create_schema(self) -> CollectionSchema:
     def describe_collection(self, name: str, **kwargs) -> CollectionInfo:
         pass
 
-    @NotImplementedError
     def drop_collection(self, name: str, **kwargs) -> None:
-        pass
+        """ Drop the collection
+
+        Args:
+            name (``str``): Specifies the name of the collection.
+            **kwargs (``dict``, optional):
+
+                * *timeout*(``float``): Specifies the timeout duration of this operation in seconds.
+                    The value defaults to ``None``, indicating that no such limit applies.
+
+        Raises:
+            MilvusException: If anything goes wrong.
+
+        Examples:
+            >>> from pymilvus import MilvusClient
+            >>> client = MilvusClient("localhost", "19530")
+            >>> client.has_collection("hello_milvus")
+            False
+        """
+        TypeChecker.check(name=name)
+        req = milvus_types.DropCollectionRequest(collection_name=name)
+        status = self.handler.DropCollection(req)
+
+        if status.error_code != common_pb2.Success:
+            raise MilvusException(status.error_code, status.reason)
 
-    @NotImplementedError
     def has_collection(self, name: str, **kwargs) -> bool:
-        pass
+        """ Check if this collection exists
+
+        Args:
+            name (``str``): Specifies the name of the collection.
+            **kwargs (``dict``, optional):
+
+                * *timeout*(``float``): Specifies the timeout duration of this operation in seconds.
+                    The value defaults to ``None``, indicating that no such limit applies.
+
+        Raises:
+            MilvusException: If anything goes wrong.
+
+        Examples:
+            >>> from pymilvus import MilvusClient
+            >>> client = MilvusClient("localhost", "19530")
+            >>> client.has_collection("hello_milvus")
+            False
+        """
+        TypeChecker.check(name=name)
+        req = milvus_types.DescribeCollectionRequest(collection_name=name)
+        resp = self.handler.DescribeCollection(req)
+
+        if resp.status.error_code == common_pb2.Success:
+            return True
+
+        # TODO: Workaround for unreasonable describe collection results and error_code
+        if resp.status.error_code == common_pb2.UnexpectedError and "can\'t find collection" in resp.status.reason:
```

Comment:
I think it'll be better that we also check if the error code is "CollectionNotExist".